cli/src/config.rs

```diff
@@ -434,6 +435,41 @@ fn read_config_path(config_path: &Path) -> Result<config::Config, config::Config
         .build()
 }
 
+#[cfg(target_os = "macos")]
+fn config_dir() -> Option<PathBuf> {
+    // Define a closure to check if config file exists in a directory
+    let check_config = |dir: &Path| {
+        let mut config_file = dir.to_owned();
+        config_file.push("jj/config.toml");
+        if config_file.exists() {
+            Some(dir.to_owned())
+        } else {
+            None
+        }
+    };
+
+    // Try getting config directory from Xdg base strategy first.
+    let path = etcetera::base_strategy::Xdg::new()
```

Comment:
We probably should report an error if both the Apple config file and the XDG config file exist. This is what we do if both `~/.jjconfig` and the Apple config file exist. It'll require people to remove one of them, but that's probably better than them thinking that they are using one file while they are actually using another one.

Reply:
Alternatively (and I'm not sure this is what we should do), we could *only* report an error if the Apple path exists, and always use the XDG path of `.jjconfig.toml`, like on Linux. I'm not sure what should happen on Windows, but that's outside the scope of this PR anyway.

Reply:
I like the idea of only having one location on MacOS in addition to `$HOME/.jjconfig.toml`. Error on finding one in Application Support, and only use the XDG one. Is that acceptable?

Reply:
I think that would work, unless somebody feels there's some advantage to following the Apple way.

You'd have to mark it as a breaking change in the changelog. We'd also want to keep a lookout for confused or unhappy users after merging that.

Reply:
Is it bad form to move the file if it exists in the old location? Like cleaning up from old versions...

Reply:
> Is it bad form to move the file if it exists in the old location? Like cleaning up from old versions...

Hm I'd find that confusing and would rather do it myself.

> You'd have to mark it as a breaking change in the changelog. We'd also want to keep a lookout for confused or unhappy users after merging that.

I'll mark it breaking.

Reply:
It might be better to have the user move it, just to make sure they know what changed. We could print the `mkdir -p && mv` command for them, though.

Reply:
I went with the last one.